iot/api-client/beta-features/manager/src/test/java/com/example/cloud/iot/examples/GatewayIT.java

```diff
@@ -0,0 +1,264 @@
+
+    // Clean up
+    DeviceGatewayExample.deleteDevice(deviceName, PROJECT_ID, CLOUD_REGION, REGISTRY_ID);
+    DeviceGatewayExample.deleteDevice(gatewayName, PROJECT_ID, CLOUD_REGION, REGISTRY_ID);
+    DeviceGatewayExample.deleteRegistry(CLOUD_REGION, PROJECT_ID, REGISTRY_ID);
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      topicAdminClient.deleteTopic(topic.getNameAsTopicName());
+    }
+  }
+
+  // MQTT device tests
+  @Test
+  public void testGatewayListenForDevice() throws Exception {
+    final String gatewayName = "rsa-listen-gateway";
+    final String deviceName = "rsa-listen-device";
+    topic = DeviceGatewayExample.createIotTopic(PROJECT_ID, TOPIC_ID);
+    DeviceGatewayExample.createRegistry(CLOUD_REGION, PROJECT_ID, REGISTRY_ID, TOPIC_ID);
+    DeviceGatewayExample.createGateway(
+        PROJECT_ID, CLOUD_REGION, REGISTRY_ID, gatewayName, RSA_PATH, "RS256");
+    DeviceGatewayExample.createDevice(PROJECT_ID, CLOUD_REGION, REGISTRY_ID, deviceName);
+    DeviceGatewayExample.bindDeviceToGateway(
+        PROJECT_ID, CLOUD_REGION, REGISTRY_ID, deviceName, gatewayName);
+
+    Thread deviceThread = new Thread() {
+      public void run() {
+        try {
+          DeviceGatewayExample.listenForConfigMessages(
+              "mqtt.googleapis.com", (short)443, PROJECT_ID, CLOUD_REGION, REGISTRY_ID, gatewayName,
+              PKCS_PATH, "RS256", deviceName);
+        } catch (Exception e) {
+          // TODO: Fail
+          System.out.println("Failure on Exception");
+        }
+      }
+    };
+    deviceThread.start();
+    Thread.sleep(3000); // Give the device a chance to connect / receive configurations
+    deviceThread.join();
+
+    // Assertions
+    String got = bout.toString();
+    System.out.println(got);
+    Assert.assertTrue(got.contains("Payload"));
+
+    // Clean up
+    DeviceGatewayExample.unbindDeviceFromGateway(
+        PROJECT_ID, CLOUD_REGION, REGISTRY_ID, deviceName, gatewayName);
+    DeviceGatewayExample.deleteDevice(deviceName, PROJECT_ID, CLOUD_REGION, REGISTRY_ID);
+    DeviceGatewayExample.deleteDevice(gatewayName, PROJECT_ID, CLOUD_REGION, REGISTRY_ID);
+    DeviceGatewayExample.deleteRegistry(CLOUD_REGION, PROJECT_ID, REGISTRY_ID);
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      topicAdminClient.deleteTopic(topic.getNameAsTopicName());
+    }
+  }
+
+  @Test
+  public void testErrorTopic() throws Exception {
+    final String gatewayName = "rsa-listen-gateway-test";
+    topic = DeviceGatewayExample.createIotTopic(PROJECT_ID, TOPIC_ID);
+    DeviceGatewayExample.createRegistry(CLOUD_REGION, PROJECT_ID, REGISTRY_ID, TOPIC_ID);
+    DeviceGatewayExample.createGateway(PROJECT_ID, CLOUD_REGION, REGISTRY_ID, 
+        gatewayName, RSA_PATH, "RS256");
+    MqttClient client = DeviceGatewayExample.startMqtt("mqtt.googleapis.com", (short)443,
+          PROJECT_ID, CLOUD_REGION, REGISTRY_ID, gatewayName,
```

Comment:
Indentation here looks weird. Make sure it's two "indents" in which looks to be ~4 spaces if you're using spacing of two per \t